### DISCUSSION TITLE
Probes do not work if UI port is switched to 443

### DISCUSSION THREAD

**Discussion by Atif-Syed-1** [datalust/seq-tickets#2000] [Help]
Hi Seq team

We have a policy of using end to end TLS for all containers.

I am deploying seq to Kubernetes (AKS cluster + traefik ingress controller). When I switch the UI port to use 443 in the config.yaml like the following

ui:
  containerPort: 443
  service:
    port: 443
    
The above fails the probes. When I looked at the Helm chart from Seq, I see the port UI is used in probes but the "scheme" value is not parameterized in helm. 
https://github.com/datalust/helm.datalust.co/blob/main/charts/seq/templates/deployment.yaml#L62

I had to manually edit the Kubernetes Seq deployment to set the following in liveness , readiness and startup probe.
scheme: HTTPS

After making the above manual tweak , my pod comes up fine with probes. I was unable to set scheme using the helm and I had to manually tweak the deployment. 

My observation:
Anyone who wants to run end to end TLS including the UI port, can't because all probe default to use the HTTP scheme. This do not works with TLS probes. Can you please have a look at the scheme parameter in the helm chart and advise if I have missed something or the scheme in probes needs to be parameterized in the helm chart from Seq.

Regards

Atif

**Comment by nblumhardt** [comment]
Thanks for raising this, it does seem as though we should expose an option here. We'll take a closer look as soon as we have a chance.

**Comment by jlegan** [comment]
@nblumhardt  is this expected though? We are migrating our Enterprise Seq license servers, which as you know we have been using goinb back to 2016, from Windows to AKS. Since everything we do is e2e TLS, in this case Seq being behind Front Door and the origin group being a private AKS cluster (private link service). Are we the only customers using health probes for containerized Seq with e2e TLS? 

When discussing it with @Atif-Syed-1 I told him I was struggling with this being necessary given the nature of data Seq typically touches and that it was not found in the discussions or issues list.

**Reply by nblumhardt** [reply]
Hi James; yes, it's a bit of a head-scratcher for me, too.

I've dug around and can't find anything interesting in our support history around this; I think it's just been very common to customize the Helm chart when deploying Seq on Kubernetes, and also fairly typical to terminate TLS at the ingress (end-to-end TLS has only been supported by the Docker build of Seq since 2021.3), so this might have been lurking but under the radar...

I'll convert this into an issue and see what we can come up with to streamline this a bit 👍
